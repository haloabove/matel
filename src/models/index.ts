export interface ListItem {
  userName: string;
  address: string;
}

export interface LogData {
    ipAddress: string;
    timestamp: string;
    method: string;
    request: string;
    statusCode: number;
    userAgent: string;
}

interface UserData {
    userName: string;
    address: string;
}

export interface ListClient {
  fetchList: (user: string) => Promise<[]>;
  fetchLocalData: () => Promise<UserData[]>;
  loadFile: () => Promise<LogData[] | null>;
}

export interface RootState {
  listItems: ListItem[];
}

export interface GettersRootState {
  listItems: ListItem[];
}