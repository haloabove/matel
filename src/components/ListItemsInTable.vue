<template>
  <b-container class="bv-example-row">
    <b-row>
      <b-col>
        <div>
          Sorting By: <b>{{ sortBy }}</b>, Sort Direction:
          <b>{{ sortDesc ? 'Descending' : 'Ascending' }}</b>
        </div>
        <b-table 
          striped
          hover
          :items="listItems"
          :fields="fields"
          :sort-by.sync="sortBy"
          :sort-desc.sync="sortDesc"
          stacked="sm"
          responsive
          ></b-table>
      </b-col>
    </b-row>
  </b-container>
</template> 

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import { Getter } from 'vuex-class';
import { LogData } from '@/models/index';

@Component({
})
export default class ListItemsInTable extends Vue {
  @Getter('listItems') private listItems!: LogData[];
    
  public sortBy = 'ipAddress';
  public sortDesc = false;
  public fields = [
    { key: 'ipAddress', sortable: true },
    { key: 'timestamp', sortable: true },
    { key: 'method', sortable: true },
    { key: 'request', sortable: false },
    { key: 'statusCode', sortable: false },
    { key: 'userAgent', sortable: false },
  ];
}
</script>

<style scoped lang="scss">

</style>
