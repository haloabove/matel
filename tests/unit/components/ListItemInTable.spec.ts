import Vue from 'vue';
import { shallowMount } from '@vue/test-utils';
import ListItemsInTable from '@/components/ListItemsInTable.vue';
import Vuex from 'vuex';
import { LogData } from '@/models/index';

Vue.use(Vuex)

// Mock Vuex store
const store = new Vuex.Store({
  state: {
    listItems: [
      { ipAddress: '168.41.191.40', timestamp: '2024-08-14T10:00:00Z', method: 'GET', request: '/api/data', statusCode: 200, userAgent: 'Mozilla/5.0' },
      { ipAddress: '177.71.128.21', timestamp: '2024-08-14T10:05:00Z', method: 'POST', request: '/api/submit', statusCode: 500, userAgent: 'Chrome/91.0' },
    ] as LogData[],
  },
  getters: {
    listItems: (state) => state.listItems,
  },
});

describe('ListItemsInTable.vue', () => {
    
  it('renders correctly', () => {
    const wrapper = shallowMount(ListItemsInTable, {
      mocks: {
        $store: store,
      },
      stubs: {
        'b-container': true,
        'b-row': true,
        'b-col': true,
        'b-table': true,
      },
    });

    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('displays sorting information', () => {
    const wrapper = shallowMount(ListItemsInTable, {
      mocks: {
        $store: store,
      },
      stubs: {
        'b-container': true,
        'b-row': true,
        'b-col': true,
        'b-table': true,
      },
    });

    expect(wrapper.find('b').text()).toContain('ipAddress');
  });
});