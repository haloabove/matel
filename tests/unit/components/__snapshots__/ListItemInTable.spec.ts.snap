// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemsInTable.vue renders correctly 1`] = `
<b-row-stub>
  <b-col-stub>
    <div>
      Sorting By: <b>ipAddress</b>, Sort Direction:
      <b>Ascending</b>
    </div>
    <b-table-stub striped="" hover="" items="[object Object],[object Object]" fields="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]" sort-by="ipAddress" stacked="sm" responsive=""></b-table-stub>
  </b-col-stub>
</b-row-stub>
`;
